System/OsString.hs

```diff
@@ -51,10 +134,73 @@ import System.OsString.Internal
     , encodeFS
     , osstr
     , pack
+    , empty
+    , singleton
     , decodeUtf
     , decodeWith
     , decodeFS
     , unpack
+    , snoc
+    , cons
+    , last
+    , tail
+    , uncons
+    , head
+    , init
+    , unsnoc
+    , null
+    , length
+    , map
+    , reverse
+    , intercalate
+    , foldl
+    , foldl'
+    , foldl1
+    , foldl1'
+    , foldr
+    , foldr'
+    , foldr1
+    , foldr1'
+    , all
+    , any
+    , concat
+    , replicate
+    , unfoldr
+    , unfoldrN
+    , take
+    , takeEnd
+    , takeWhileEnd
+    , takeWhile
+    , drop
+    , dropEnd
+    , dropWhileEnd
+    , dropWhile
+    , break
+    , breakEnd
+    , span
+    , spanEnd
+    , splitAt
+    , split
+    , splitWith
+    , stripSuffix
+    , stripPrefix
+    , isInfixOf
+    , isPrefixOf
+    , isSuffixOf
+    , breakSubstring
+    , elem
+    , find
+    , filter
+    , partition
+    , index
+    , indexMaybe
+    , (!?)
+    , elemIndex
+    , elemIndices
+    , count
+    , findIndex
+    , findIndices
     )
 import System.OsString.Internal.Types
     ( OsString, OsChar )
+import Prelude hiding (last, tail, head, init, null, length, map, reverse, foldl, foldr, foldl1, foldr1, all, any, concat, replicate, take, takeWhile, drop, dropWhile, break, span, splitAt, elem, filter)
```

Comment:
This will break when `base-4.20` adds `foldl'` to `Prelude`;  the [migration guide](https://github.com/haskell/core-libraries-committee/blob/9adab1c0e455d50e9210335edc2d33d117f2a086/guides/export-foldl-prime-prelude.md) suggests to hide `Foldable(..)` rather than its individual methods.

(Here `import Prelude ()` is also pretty clear.)